test/mini_racer_test.rb

```diff
@@ -448,29 +450,33 @@ def test_an_empty_snapshot_is_valid
     GC.start
   end
 
-  def test_snapshots_can_be_warmed_up_with_no_side_effects
-    # shamelessly inspired by https://github.com/v8/v8/blob/5.3.254/test/cctest/test-serialize.cc#L792-L854
-    snapshot_source = <<-JS
-      function f() { return Math.sin(1); }
-      var a = 5;
-    JS
+# segfault
+# CFUNC  :warmup_unsafe!
+# METHOD /Users/loic.nageleisen/Source/github.com/rubyjs/libv8-node/test/mini_racer/lib/mini_racer.rb:452 (def warmup!)
```

Comment:
`warmup_unsafe!` indeed! Either we don't do something that we should to make it safe, or we pass something that was okay before and is not now.

```
    def warmup!(src)
      # we have to do something here
      # we are bloating memory a bit but it is more correct
      # than hitting an exception when attempty to compile invalid source
      begin
        ctx = MiniRacer::Context.new
        ctx.eval(@source)
        ctx.eval(src)
      rescue MiniRacer::RuntimeError => e
        raise MiniRacer::SnapshotError.new, e.message, e.backtrace
      end

      warmup_unsafe!(src)
    end
```